test/js-native-api/test_object/test_object.c

```diff
@@ -629,12 +629,21 @@ TypeTaggedInstance(napi_env env, napi_callback_info info) {
   size_t argc = 1;
   uint32_t type_index;
   napi_value instance, which_type;
+  napi_type_tag tag;
+
+  // Below we copy the tag before setting it to prevent bugs where a pointer
+  // to the tag (instead of the 128-bit tag value) is stored.
+  // To make sure this kind of bug is detected, memset the tag copy to 0.
 
   NODE_API_CALL(env, napi_get_cb_info(env, info, &argc, &which_type, NULL, NULL));
   NODE_API_CALL(env, napi_get_value_uint32(env, which_type, &type_index));
   VALIDATE_TYPE_INDEX(env, type_index);
   NODE_API_CALL(env, napi_create_object(env, &instance));
-  NODE_API_CALL(env, napi_type_tag_object(env, instance, &type_tags[type_index]));
+  tag = type_tags[type_index];
+  NODE_API_CALL(env, napi_type_tag_object(env, instance, &tag));
+
+  // Make sure tag value is stored and not just a pointer
```

Comment:
The memset is unnecessary because `tag` will go out of scope so, if the API doesn't copy it, the test will crash.

Reply:
Actually (unless some strong stack guards or sanitizers are used for tests), the stack memory will most likely still be R+W when the check occurs. So some data will be there.

It's very likely to have changed due to subsequent calls, so the memset might be overkill. But the stack memory at the address of the tag could also remain unchanged (unless subsequent functions calls have clobbered it), and so a pointer would still point to the correct tag, by accident.

That's why I used memset, although I did not make it clear in the comment. So I can either improve the comment or remove the memset if you think it's better.

Reply:
OK. Let's keep the memset, but please update the comment!